### PR TITLE
Step1-3

### PR DESCRIPTION
**Pull request by mkruckow**
I have restructured the repo:
- There is a common_code, which contains:
  - [x] The needed files from the in-official MESA branch on ionization;
    - incl. ionization_data
  - [x] The CE code converted to the requirements of the new MESA version 23.05.1
  - [x] Common functions of `run_star_extras.f90` for the CE part (steps 2 and 3) moved to `CE_run_star_extras.f90`
  - [x] Common code from the `run_star_extras.f90` and `run_binary_extras.f90` moved to `POSYDON_run_star_extras.f90` and `POSYDON_run_binary_extras.f90` (step 1)
- There are common inlists
  - [x] The inlists are updated to the current MESA version 23.05.1, which Jaime and me are using.
  - [x] The inlists got commented with the meaning of the variable, their MESA default value and a comment about why we choose that value.
  - [x] One for the star module
  - [x] One for the binary module
  - [x] There are common controls
    - for the first step(s): `inlist_x_ctrls_POSYDON`
    - for the later steps: `inlist_x_ctrls_CE`
  - [x] There are common history and profile columns lists
  - In the future we may want to have some common plotting.
- [x] There is a directory to contain the models
  - It has the ZAMS file in there and the models of the steps are supposed to be copied there as a resource for the next step.
  - [x] There is a new `cpy` script in each step to copy the output of the step to the `models` directory.
- [x] There are some overall scripts, which we can use to run several steps in a row.
  - `rn_HMS-CO.sh`
  - `rn_HMS-HMS.sh`
  - `rn_HMS-single.sh`
- [x] The make files are adopted to compile the common code, where needed.
- [x] The individual inlists are shortened to contain only needed stuff and having files to only take care of correct inlist stacking.
- The old step1 got split into two:
  - `step1_HMS-CO`: it evolves only one star and a point mass (will be mainly used by me)
  - `step1_HMS-HMS`: it evolves two HMS stars in parallel (will be mainly used by Jaime)
  - `step1_HMS-single`: it evolves a HMS star in isolation until a certain radius is reached (will be mainly used by Jaime)
  - we can potentially add more, e.g. for the planet engulfment
  - [x] Changes in `run_star_extras.f90`
    - Add back the `get_ion_info` function
    - Uncomment not needed terminal outputs
  - [x] Changes to `run_binary_extras.f90`
    - There was a typo: isnan -> is_nan
    - I added a new column in the binary history: `adjust_J_q` because of a more detailed inspection of the angular momentum adjustment in the test runs; similarly I added an output for another CE initialization case: Exceeded photon trapping radius
    - Fix a bug: The check for a point mass needs to be done before accessing the star's values.
- Migrating step2:
  - no longer supported inlist variables are removed
  - no longer supported hooks are removed
  - [x] It compiles
  - [x] It runs
- Migrating step3:
  - no longer supported inlist variables are removed
  - no longer supported hooks are removed
  - [x] It compiles
  - [x] It runs

### CODE REVIEW

File: step1_HMS-CO/inlist1 [hunk 1]
```diff
@@ -0,0 +1,71 @@
+&star_job
+
+    read_extra_star_job_inlist(1) = .true.
+    extra_star_job_inlist_name(1) = '../inlists/inlist_star_common'
+
+    read_extra_star_job_inlist(2) = .true.
+    extra_star_job_inlist_name(2) = 'inlist_star_step1'
+
+    read_extra_star_job_inlist(3) = .true.
+    extra_star_job_inlist_name(3) = 'inlist_star1'
```
Comment (tassos25) [inline step1_HMS-CO/inlist1:10]: What is the reason for adding the third layer of insists

File: step1_HMS-CO/inlist_star_step1 [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+!!! opacities
+
+/ ! end of kap namelist
+
+&controls
+
+!!! output controls
+   terminal_interval = 10       ! interval for terminal outputs; common value: 1        | get less outputs
+
+!!! mass gain or loss
```
Comment (tassos25) [inline step1_HMS-CO/inlist_star_step1:27]: Can't we just remove this whole section of wind parameters, as we are using the POSYDON default ones?

File: step1_HMS-CO/inlist_star_step1 [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+   hot_wind_scheme = 'Dutch'         ! hot wind scheme: hot of all stars;                        common value: '' |
+   cool_wind_RGB_scheme = 'Dutch'    ! cool wind scheme: cool of high mass stars (Minit>10Msun); common value: '' | it gets replaced in run_star_extras by 'Reimers' for cool low mass stars
+   cool_wind_AGB_scheme = 'Blocker'  ! cool wind scheme: cool for low mass stars in AGB;         common value: '' |
+
+!!! timestep controls (limits reduce next timestep, hard limits causes retry)
+   solver_iters_timestep_limit = 20  ! maximum number of solver iterations before timestep gets reduced; common value: 7 | try to avoid too many timestep changes
+
+!!! miscellaneous controls
+ !! hooks
+   use_other_wind = .true.  ! enable other wind prescription; common value: .false. | combination of Dutch and Reimers, with Z=Zbase for the scaling
```
Comment (tassos25) [inline step1_HMS-CO/inlist_star_step1:37]: Same for this one, isn't this turn on by default in teh default POSYDON?

**Review by tassos25 — APPROVED** [review]
An amazing, ginormous PR!

File: step1_HMS-CO/inlist1 (hunk 1 above)
Comment (mkruckow) [reply step1_HMS-CO/inlist1:10]: The third layer makes the naming, hence this is easier to understand for the HMS-HMS case.

File: step1_HMS-CO/inlist_star_step1 (hunk 2 above)
Comment (mkruckow) [reply step1_HMS-CO/inlist_star_step1:27]: This part here exactly puts the POSYDON default for step 1.

File: step1_HMS-CO/inlist_star_step1 (hunk 3 above)
Comment (mkruckow) [reply step1_HMS-CO/inlist_star_step1:37]: This part here exactly puts the POSYDON default for step 1.

**Pull request merged**